### PR TITLE
Use YAML literal block syntax for description, rationale

### PR DESCRIPTION
**Pull request by zcorpan**
This prevents some syntax issues with YAML, e.g. '#' can be used in a literal block without being treated as a line comment.

The switch from PyYAML to ruamel.yaml was needed to validate the syntax style.

### CODE REVIEW

File: activities.py [hunk 1]
```diff
@@ -1,150 +1,136 @@
...
-            self.log_error(f"'id' must be a string without whitespace, got {item['id']}", key, line)
-        if 'description' in item and item['description'] not in [None, '']:
-            if not isinstance(item['description'], str) or not item['description'].strip():
-                self.log_error(f"'description' must be a string or null, got {item['description']}", key, line)
+            self.log_error(f"'id' must be a string without whitespace, got {item['id']}.", key)
+
         if 'bug' in item and item['bug'] not in [None, '']:
             if not isinstance(item['bug'], str) or not item['bug'].startswith("https://bugzilla.mozilla.org/show_bug.cgi?id="):
-                self.log_error(f"'bug' must be a URL starting with 'https://bugzilla.mozilla.org/show_bug.cgi?id=', got {item['bug']}", key, line)
+                self.log_error(f"'bug' must be a URL starting with 'https://bugzilla.mozilla.org/show_bug.cgi?id=', got {item['bug']}.", key)
```
Comment (martinthomson) [inline activities.py:85]: If we're requiring this, why not include just the bug number (and take the `https://bugzilla...` as implied?)

Not for this change, but as a followup perhaps.

File: activities.py [hunk 2]
```diff
@@ -1,150 +1,136 @@
...
+
         if 'caniuse' in item and item['caniuse'] not in [None, '']:
             if not isinstance(item['caniuse'], str) or not item['caniuse'].startswith("https://caniuse.com/"):
-                self.log_error(f"'caniuse' must be a URL starting with 'https://caniuse.com/' or empty, got {item['caniuse']}", key, line)
+                self.log_error(f"'caniuse' must be a URL starting with 'https://caniuse.com/', got {item['caniuse']}.", key)
```
Comment (martinthomson) [inline activities.py:89]: As above.

File: activities.py [hunk 3]
```diff
@@ -1,150 +1,136 @@
...
+
+        if 'mdn' in item and item['mdn'] not in [None, '']:
+            if not isinstance(item['mdn'], str) or not item['mdn'].startswith("https://developer.mozilla.org/en-US/"):
+                self.log_error(f"'mdn' must be a URL starting with 'https://developer.mozilla.org/en-US/', got {item['mdn']}.", key)
```
Comment (martinthomson) [inline activities.py:93]: As above

File: activities.py [hunk 4]
```diff
@@ -1,150 +1,136 @@
...
+
+    def validate_literal_block(self, data: CommentedMap, key: str, field: str):
+        value = data.get(key, {}).get(field)
+        if value is not None:
+            if not isinstance(value, LiteralScalarString):
+                if self.fix:
+                    data[key][field] = LiteralScalarString(value)
+                else:
+                    self.log_error(f"'{field}' must use literal block syntax (|).", key)
+            elif not value.endswith('\n'):
```
Comment (martinthomson) [inline activities.py:46]: Is it even possible to have a value that doesn't have a trailing `\n`?

File: activities.yml [hunk 5]
```diff
@@ -1,2989 +1,2479 @@
 <link>'s imagesrcset and imagesizes attributes:
   bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1515760
-  caniuse: null
-  description: Adds imagesrcset and imagesizes attributes to &lt;link> which correspond
-    to the srcset and sizes attributes of &lt;img> respectively, for the purposes
-    of preloading.
+  caniuse:
```
Comment (martinthomson) [inline activities.yml:3]: We've a bunch of these empty values.  I assume that these can be dropped from the file safely.  Should they be?

File: merge-data.py [hunk 6]
```diff
@@ -13,11 +17,19 @@
...
-        if dict1[key]:
-            dict2[key] = dict1[key]
-    dict2.pop('issue', None)
+    """
+    Merges data from dict1 into dict2. Keys in dict1 take precedence if not None.
+    Strips trailing newline from 'description' and 'rationale'.
+    """
+    for key, value in dict1.items():
+        if value is not None:
+            if key in ['description', 'rationale'] and isinstance(value, str):
```
Comment (martinthomson) [inline merge-data.py:26]: Do you need to check for description/rationale?  Could this trim all strings?

**Review by martinthomson — APPROVED** [review]

File: activities.py (hunk 4 above)
Comment (zcorpan) [reply activities.py:46]: Yes:
```
foo: |-
  bar
```

File: activities.py (hunk 1 above)
Comment (zcorpan) [reply activities.py:85]: I think it's easier for contributors to paste the full URL for all of the URLs than to find out what substring should be used. Though new issues will have the URLs in the issue OP instead, so it doesn't matter so much.

File: merge-data.py (hunk 6 above)
Comment (zcorpan) [reply merge-data.py:26]: It could trim all strings.

**Pull request merged**